acceptance/tests/utf8/utf8-in-file-resource.rb

```diff
@@ -0,0 +1,55 @@
+test_name 'utf-8 characters in resource title and param values' do
+  confine :except, :platform => [ 'windows', 'ubuntu-16']   # PUP-6983
+  utf8chars = "€‰ㄘ万竹ÜÖ"
+  agents.each do |agent|
+    agent_file = agent.tmpfile("file" + utf8chars) 
+    # remove this file, so puppet can create it and not merely correct
+    # its drift.
+    on(agent, "rm -rf #{agent_file}")
+
+    manifest =
+<<PP
+
+file { "#{agent_file}" :
+  ensure => file,
+  mode => "0644",
+  content => "This is the file content. file #{utf8chars} 
+",
+}
+
+PP
+
+    step "Apply manifest" do
+      apply_manifest_on(
+        agent,
+        manifest,
+        {:acceptable_exit_codes => [0, 2], :catch_failures => true, }
```

Comment:
some of our vmpooler acceptance hosts don't have their locale set to utf-8 by default, which (has and) will cause myriad issues with these utf-8 chars. To ensure the behavior we're checking for, until we've fixed it so that puppet on *nix handles utf-8 encoding even when its environment is not utf-8, we need to specify the encoding via environment here. An example is `acceptance/tests/resource/user/utf8_user_comments.rb`, added here: https://github.com/puppetlabs/puppet/pull/5419/files